app/services/invite_schools.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class InviteSchools
+  def run(school_ids)
```

Comment:
I might be missing some context, but...

How many times do we expect it to run, and with how many schools exactly?

If we fire it off with 20k schools, it will `deliver_now` 20k emails one after another (one request after another), and fire off some 40k read queries and 20k write queries to the database. 

Queries we can probably handle, I hope. We kind of do that in school import. But firing off that many requests to notify in a short time sounds scary. Do they have some 'send in bulk' option? (It's scary because 1. notify might not like it and 2. because deliver_now is synchronous, I think, and making 20k api calls to the internet will take a looong time).

Actually, I checked it out. Their send in bulk option is quite crap. Manually upload all the email addresses to notify and send it. Ugh.

Do we worry about the load at all? Can we even load test this without spamming schools?

Reply:
All legitimate concerns you raise. Quite happy to look into how we can space out the sending of the emails using delayed job. I would do that in a separate pr though.

Reply:
@kicferk1 forgot to add that from what product have said, they will tell us the list of schools to send to or perhaps the criteria of which schools we send to in order to avoid overwhelming our infrastructure. Until, those requirements become clear it's difficult to be exact with the numbers

Reply:
My understanding is we'll probably have another rake task that selects the schools we want to send to and then calls this one.

Reply:
Alright, I think it would be convenient to explore options of handling this amount of emails - perhaps other DfE services need to deal with something similar? I am quite happy for it to live in its own PR. 

After all, if we end up with something that can handle all schools at once, it might save our team a lot of time and effort coming up with batches and triggering it for all of those batches.